<launch>
 <!-- Joystick node -->
  <node respawn="true" pkg="joy" type="joy_node" name="robot_joy">
    <param name="dev" type="string" value="/dev/input/js0"/>
    <param name="deadzone" value="0.12"/>
  </node>

 <!-- Teloperation node -->
  <param name="axis_linear" value="1" type="int"/>
  <param name="axis_angular" value="0" type="int"/>
  <param name="scale_linear" value="1" type="double"/>
  <param name="scale_angular" value="1" type="double"/>
  <node pkg="learning_joy" type="robot_teleop_joy" name="teleop"/>

 <!-- Collision avoidance node -->
  <node pkg="collision_avoidance" type="collision_avoidance" name="coll_avoid" output="screen">
      launch-prefix="xterm -e gdb --args"
      <param name="radius_max" value="1.2"/>
      <param name="radius_min" value="0.5"/>
      <param name="alpha_max" value="20.0"/>
      <remap from="~scans" to="/vrep/hokuyoSensor"/>
      <remap from="~cmd_vel" to="cmd_vel"/>
      <remap from="~output_vel" to="/vrep/twistCommand"/>
  </node>
  
	<node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name ="pointcloud_to_laserscan_node" output="screen">
	 
	 <remap from="cloud_in" to="/vrep/hokuyoSensor"/>
	</node>

	<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen" >

		<!--<remap from /scan to scan> -->
		 
		<param name="base_frame" value="rosControlledBubbleRob"/>
		<param name="odom_frame" value="/world"/>

	</node>
    
</launch>
